master/internal/api_config_policies.go

```diff
@@ -0,0 +1,118 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/protobuf/types/known/structpb"
+	"gopkg.in/yaml.v3"
+
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+// workload types are used to distinguish configuration for experiments and NTSCs.
+const (
+	// Should not be used.
+	WorkloadTypeUnspecified = "UNKNOWN"
+	// Configuration policies for experiments.
+	WorkloadTypeExperiment = "EXPERIMENT"
+	// Configuration policies for NTSC.
+	WorkloadTypeNTSC = "NTSC"
+)
+
+func validWorkloadEnum(val string) bool {
+	switch val {
+	case WorkloadTypeExperiment, WorkloadTypeNTSC:
+		return true
+	default:
+		return false
+	}
+}
+
+var yamlString = `
+invariant_configs:
+  description: "test"
+constraints:
+  resources:
+    max_slots: 4
+  priority_limit: 10
+ `
+
+func stubData() (*structpb.Struct, error) {
+	// put yaml string into a map
+	var yamlMap map[string]interface{}
+	if err := yaml.Unmarshal([]byte(yamlString), &yamlMap); err != nil {
+		return nil, fmt.Errorf("unable to unmarshal yaml: %v", err)
+	}
+
+	// convert map to protobuf struct
+	yamlStruct, err := structpb.NewStruct(yamlMap)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal to struct: %v", err)
+	}
+	return yamlStruct, nil
+}
+
+// Add or update workspace task config policies.
+func (*apiServer) PutWorkspaceConfigPolicies(
+	ctx context.Context, req *apiv1.PutWorkspaceConfigPoliciesRequest,
+) (*apiv1.PutWorkspaceConfigPoliciesResponse, error) {
+	if !validWorkloadEnum(req.WorkloadType) {
```

Comment:
I like this `validWorkloadEnum` as a function rather than something like a `.Valid()` method on a `WorkloadType` type. Good call.

Reply:
What are the benefits to doing it this way as opposed to the `.Valid()` method type? Just curious!

Reply:
In my opinion, this way is simpler and more general. To define a `.Valid()` method, we would also need to define a `WorkloadType` ... which is fine, but it's unnecessary at this point in development. 

My understanding of golang is that it's not an object-oriented language, although we can do a lot of things that feel like object-oriented programming. Golang favors simplicity and fewer types. With Golang, my default is *not* to use methods unless I have a compelling reason to use one. In contrast, I think with a language like Java the default would be to create an object for everything but the most basic primitives (and sometimes even then too).

@maxrussell please feel free to correct me or add your own reasoning!

Reply:
In this case, I think we would also have to first convert the request input into a `WorkloadType` to use a `.Valid()` method?

Reply:
@kkunapuli: No, you nailed it: that's *exactly* my reasoning/preference!

Scanning a JSON string into a field of [Kind](https://pkg.go.dev/reflect#Kind) `string` works just fine for those situations, but you're absolutely correct, Kristine, that if we e.g. had a `string` from gRPC Gateway or [something like this](https://go.dev/play/p/j9QoFy-s2VB), we'd first have to convert to call the method, which is ugly.

Reply:
_Ooh_ I see, this makes sense, thanks guys! The example was really helpful, and I didn't know that Go had a `Kind` type! 
Also, really good point about only defining a type if it provides an inherent purpose that can't be accomplished by a standalone function!

Reply:
I didn't know about `Kind` either! TIL